src/controller/net/AutoRefServer.java

```diff
@@ -0,0 +1,81 @@
+package controller.net;
+
+import controller.SystemClock;
+import controller.action.ActionBoard;
+
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.SynchronousQueue;
+
+public class AutoRefServer implements Runnable
+{
+    Socket connectionSocket;
+    BlockingQueue<String> commandQueue;
+
+    BlockingQueue<String> returnCommunicationQueue;
+
+    private final static int port = 8750;
+
+    public AutoRefServer(BlockingQueue<String> commandQueue, BlockingQueue<String> returnCommunicationQueue){
+        try{
+            this.commandQueue = commandQueue;
+            this.returnCommunicationQueue = returnCommunicationQueue;
+        }catch(Exception e){
+            e.printStackTrace();
+        }
+    }
+
+    public void run() {
+        ServerSocket simulator_socket = null;
+        while (true) {
+            boolean isConnected;
+            try {
+                if (simulator_socket != null && connectionSocket != null && simulator_socket.isBound()) {
+                    connectionSocket.close();
+                    simulator_socket.close();
+                }
+                simulator_socket = new ServerSocket(port);
+                this.connectionSocket = simulator_socket.accept();
+                isConnected = true;
+                System.out.println("Client Got Connected  ");
```

Comment:
MAybe put those print statements behind a debug flag so they can be turned on/off

Reply:
Future work.